exporter/stackdriverexporter/stackdriver.go

```diff
@@ -199,7 +206,11 @@ func newStackdriverMetricsExporter(cfg *Config, params component.ExporterCreateP
 		Timeout: cfg.Timeout,
 	}
 
-	copts, err := generateClientOptions(cfg, params.ApplicationStartInfo.Version)
+	if cfg.UserAgent != "" {
```

Comment:
Inside of the `generateClientOptions` function above we have a very similar code:
```go
	if cfg.UserAgent != "" {
		copts = append(copts, option.WithUserAgent(cfg.UserAgent))
	}
```
Why can't we combine the two and make both changes in one place again? Is this block unique to metric exporter?

Also, can we only use one or another, rather than setting user agent in multiple places at once?
My understanding is that only one of them will ultimately win?

Reply:
> Why can't we combine the two and make both changes in one place again? Is this block unique to metric exporter?

Yes it's unique to the metric exporter.

> Also, can we only use one or another, rather than setting user agent in multiple places at once?
> My understanding is that only one of them will ultimately win?

- Line 210: Metrics exporter. The OC stackdriver metrics exporter always overrides the user agent using this value (or a default string if empty), so this line takes effect
- Line 75: Trace exporter. The OT operations trace exporter doesn't override the gRPC user agent so this will still impact the user agent sent on the trace exporter. This might not really be necessary but I thought it best to leave as is for now
- Line 82: Both exporters if `endpoint = local` & `use_insecure = true`